src/main/java/org/opensearch/neuralsearch/processor/chunker/FixedTokenLengthChunker.java

```diff
@@ -0,0 +1,159 @@
+        }
+
+        List<String> tokens = tokenize(content, tokenizer, maxTokenCount);
+        List<String> passages = new ArrayList<>();
+
+        String passage;
+        int startToken = 0;
+        int overlapTokenNumber = (int) Math.floor(tokenLimit * overlapRate);
+        // overlapTokenNumber must be smaller than the token limit
+        overlapTokenNumber = Math.min(overlapTokenNumber, tokenLimit - 1);
+
+        while (startToken < tokens.size()) {
+            if (startToken + tokenLimit >= tokens.size()) {
+                // break the loop when already cover the last token
+                passage = String.join(" ", tokens.subList(startToken, tokens.size()));
+                addPassageToList(passages, passage, maxChunkLimit);
+                break;
+            } else {
+                passage = String.join(" ", tokens.subList(startToken, startToken + tokenLimit));
+                addPassageToList(passages, passage, maxChunkLimit);
+            }
+            startToken += tokenLimit - overlapTokenNumber;
+        }
+        return passages;
+    }
+
+    private void addPassageToList(List<String> passages, String passage, int maxChunkLimit) {
+        if (maxChunkLimit != DEFAULT_MAX_CHUNK_LIMIT && passages.size() >= maxChunkLimit) {
+            throw new IllegalStateException("Exceed max chunk number: " + maxChunkLimit);
+        }
+        passages.add(passage);
+    }
+
+    private void validatePositiveIntegerParameter(Map<String, Object> parameters, String fieldName) {
+        // this method validate that parameter is a positive integer
+        // this method accepts positive float or double number
+        if (!parameters.containsKey(fieldName)) {
+            // all parameters are optional
+            return;
+        }
+        if (!(parameters.get(fieldName) instanceof Number)) {
+            throw new IllegalArgumentException(
+                "fixed length parameter [" + fieldName + "] cannot be cast to [" + Number.class.getName() + "]"
+            );
+        }
+        if (((Number) parameters.get(fieldName)).intValue() <= 0) {
+            throw new IllegalArgumentException("fixed length parameter [" + fieldName + "] must be positive");
+        }
+    }
+
+    @Override
+    public void validateParameters(Map<String, Object> parameters) {
+        validatePositiveIntegerParameter(parameters, TOKEN_LIMIT_FIELD);
+        validatePositiveIntegerParameter(parameters, MAX_CHUNK_LIMIT_FIELD);
+        validatePositiveIntegerParameter(parameters, MAX_TOKEN_COUNT_FIELD);
+
+        if (parameters.containsKey(OVERLAP_RATE_FIELD)) {
+            if (!(parameters.get(OVERLAP_RATE_FIELD) instanceof Number)) {
+                throw new IllegalArgumentException(
+                    "fixed length parameter [" + OVERLAP_RATE_FIELD + "] cannot be cast to [" + Number.class.getName() + "]"
+                );
+            }
+            if (((Number) parameters.get(OVERLAP_RATE_FIELD)).doubleValue() < 0.0
```

Comment:
Can we use BigDecimal for these float values? https://www.baeldung.com/java-double-vs-bigdecimal#:~:text=The%20BigDecimal%20class%20offers%20better,accurate%20than%20floating%2Dpoint%20arithmetic.

Reply:
Thanks. Will be changed.

Reply:
Per our discussion, we will keep the double implementation. I will mark this conversation as resolved.